### PR TITLE
feat: Reuse Dashboard redux data in Explore

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
Reopen https://github.com/apache/superset/pull/20668

Currently when user open Explore, we always send a request to `/v1/explore` endpoint, which assembles Explore initial data - slice, form_data and datasource.
When user goes to Explore from Dashboard (by clicking "Edit chart" or chart title), most of the data needed to assemble Explore should already be available in Redux store.
This PR implements using data from Dashboard Redux store to initialize Explore without calling `/v1/explore` endpoint.

If user opens Explore from Dashboard, check if slice, formData, datasourceId and datasourceType are available. If they are, fetch datasource metadata and hydrate Explore using slice, formData and datasource.
If they are not available, call `/v1/explore` and hydrate Explore.

We can't reuse datasource from Dashboard. Datasources metadata on Dashboard is trimmed so that datasource contains only columns and metrics used by charts on that dashboard. In Explore we need to be able to access all columns and metrics, which means we must fetch those from API.

### TESTING INSTRUCTIONS
1. Go to dashboard
2. Click "Edit chart"
3. Verify that Explore has opened correctly and that we did not call `/v1/explore/` endpoint
4. Enter Explore from any other entry point (for example Charts list or Welcome page)
5. Verify that we do call `/v1/explore`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
@jinghua-qa Could you please help with testing? This PR was previously [reverted](https://github.com/apache/superset/pull/20689) due to a bug that went unnoticed, I'm hoping to avoid repeating this mistake.

File: superset/charts/schemas.py [hunk 1]
```diff
@@ -165,6 +170,11 @@ class ChartEntityResponseSchema(Schema):
     slice_url = fields.String(description=slice_url_description)
     certified_by = fields.String(description=certified_by_description)
     certification_details = fields.String(description=certification_details_description)
+    is_managed_externally = fields.Boolean(
+        description=is_managed_externally_description
+    )
+    owners = fields.List(fields.Nested(UserSchema))
```
Comment (zhaoyongjie) [inline superset/charts/schemas.py:176]: a minor suggestion from recent fixing. the `owners` fields suggest that align with [Dashboard](https://github.com/apache/superset/pull/20826/files#) field to `owners = fields.List(fields.Dict())`.

Comment (kgabryje) [reply superset/charts/schemas.py:176]: Hmm, I'm getting a 500 from the dashboard endpoint with that change
image

**Comment by kgabryje** [comment]
I decided to close this PR - the performance gain from reusing the dashboard redux data was negligible at the cost of increased code complexity, which introduces a greater risk of bugs and regressions.

**Pull request closed without merging**